compiler/noirc_evaluator/src/acir/arrays.rs

```diff
@@ -968,16 +970,15 @@ impl Context<'_> {
         source: BlockId,
         array_len: usize,
         value_types: &[NumericType],
-    ) -> Result<im::Vector<AcirValue>, RuntimeError> {
-        let init_values = try_vecmap(0..array_len, |i| {
+    ) -> impl Iterator<Item = Result<AcirValue, RuntimeError>> {
```

Comment:
It seems we never use this iterator and always call `collect` afterwards, so I don't know if this change is worth it.

Reply:
https://github.com/noir-lang/noir/blob/0e378195dc427dfb4f169e9abd9901f05d412502/compiler/noirc_evaluator/src/acir/mod.rs#L877-L880